src/halmos/mapper.py

```diff
@@ -76,6 +79,55 @@ def __call__(cls, *args, **kwargs):
         return cls._instances[cls]
 
 
+class BuildOut(metaclass=SingletonMeta):
+    def __init__(self):
+        self._build_out_map: dict = None
+        self._build_out_map_reverse: dict = None
+
+    def set_build_out(self, build_out_map: dict):
+        if self._build_out_map is build_out_map:
+            return
+
+        self._build_out_map = build_out_map
+        self._build_out_map_reverse = None
+
+    def create_build_out_map_reverse(self):
+        # create reverse mapping
+        self._build_out_map_reverse = defaultdict(dict)
+        for filename, file_map in self._build_out_map.items():
+            for contract_name, contract_map in file_map.items():
+                self._build_out_map_reverse[contract_name][filename] = contract_map
+
+    def get_by_name(self, contract_name: str, filename: str = None) -> dict:
+        """
+        Return the build output json for the given contract name.
+
+        Raise a HalmosException if the contract is not found, or cannot be uniquely determined.
+
+        The optional filename argument is required, if the contract name appears in multiple files.
+        """
+        if not self._build_out_map_reverse:
+            self.create_build_out_map_reverse()
+
+        mapping = self._build_out_map_reverse[contract_name]
+
+        if not mapping:
+            raise HalmosException(f"{contract_name} is not found")
+
+        if filename is None:
```

Comment:
filename could be empty string

Reply:
what would be the meaning of that? isn't it just going to fail with `"{contract_name} is not found in ''"`?

Reply:
yes, that's why this branch shouldn't be taken when filename is empty. otherwise, confusing error messages may be generated.

Reply:
i mean, i want to avoid that this error message `"{contract_name} exists in multiple files: {list(mapping.keys())}"` may be printed when filename is empty.